### DISCUSSION TITLE
Postman and API Testing with Next-Auth

### DISCUSSION THREAD

**Discussion by Nish7** [nextauthjs/next-auth#2064] [Help]
## Question 💬

I am currently using next-auth for oauth (2.0) for google sign in, and I was wondering how does one do nextjs routes API testing using postman/insominia?  

## Problem:

I have a protected API route in nextjs and it is implemented using this:

let's suppose the API path is `http://localhost:3000/api/protected` and with method `POST` which returns a JSON

```
const session = await getSession({ req });
	if (session) {
                 //Session Verified
                 res.json({data:'verified'})
	} else {
		res.json({data:'Please Sign in!'});
	}

    res.end()
```
So, whenever I make a POST request (using postman) to the API route, I will get as expected `data:'Please sign in'`  
but now how should I enable testing authorisation on this route with google oauth. 

I made attempt to get the access token using these settings, however i am not sure what to do next and how does next-auth callback handles the session, To be able to test and have access to session in protected API routes in postman.

Postman oauth settings:
<img width="983" alt="Screenshot 2021-05-25 at 3 35 59 PM" src="https://user-images.githubusercontent.com/22117584/119480065-eae72080-bd6e-11eb-9c4d-edfeb1d74663.png">

When setting the `callback URL` to `http://localhost:3000/api/auth/callback/google`, it resulted in consent screen displaying 'This browser or app may not be secure.'

**Comment by balazsorban44** [comment]
I would recommend checking out [Mock Service Worker](https://mswjs.io/) to mock the entire API call to `/api/protected`

A nice relevant article: https://kentcdodds.com/blog/stop-mocking-fetch/
